### PR TITLE
EOS-21162: Add performance testing notes; add helper scripts for full fake ADDB testing

### PR DESCRIPTION
**Pull request by dsurnin**
Signed-off-by: Dmitrii Surnin <dmitrii.surnin@seagate.com>

-----
[View rendered docs/notes.md](https://github.com/Seagate/cortx-s3server/blob/br/ds/notes/docs/notes.md)

### CODE REVIEW

File: docs/notes.md [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+- example of `raw_data_dir` folder structure
+    - folder `raw_data_dir`
+        - subfolder `addb_c1_s16Mb_cnt1000`
+            - content `dumps_<PID>.txt`
+            - content `dumpc_<PID>.txt`
+        - subfolder `addb_c22_s512Kb_cnt1000`
+            - content `dumps_<PID>.txt`
+            - content `dumpc_<PID>.txt`
+
+- it is possible to add S3Authserver logs - `app.log` - to DB
```
Comment (DPG17) [inline docs/notes.md:112]: Can you provide full command to add S3AuthServer log to DB?

File: docs/notes.md [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+
+- generate histograms
+
+```
+./hist.py -t s3_request_state -u ms -r 2 [[Action::check_authentication,S3Action::load_metadata],[S3Action::load_metadata,S3Action::check_authorization],[S3Action::check_authorization,S3GetObjectAction::validate_object_info]] -o hist_auth.png -f png -s 20 10 --db m0play_4k_auth.db
```
Comment (DPG17) [inline docs/notes.md:117]: Please explain the parameters passed to "hist.py"

File: docs/notes.md [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+    - skiping different types or tests
+- run several instancies of s3server
+- run basic tests - do not run ut/st. with help of s3cmd run following scenario
+    - Create Buckets
+    - List Buckets
+    - Put Object of various sizes
+    - Get Object the uploaded objects
+    - Delete Object
+    - Delete Bucket
+- generate valgrind memleak reports
```
[the remark this answers is not in the document]
Comment (dsurnin) [reply docs/notes.md]: done

File: docs/notes.md (hunk 2 above)
Comment (dsurnin) [reply docs/notes.md:115]: Detailed explanations could be found in help output

File: docs/notes.md (hunk 1 above)
Comment (dsurnin) [reply docs/notes.md]: it looks like in the latest version it is not supported, so deleting it from notes

**Review by t7ko-seagate — APPROVED** [review]

File: run_addb_batch.sh [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+addb_dir="/var/log/addb_res/"
+s3bpath="../s3bench/s3bench"
+
+s3bpath=$(realpath -e "$s3bpath")
+mkdir -p "$addb_dir"
+
+test_plan=$(cat <<-EOF
+[
+{"clients": 1, "size": "100b", "reads": 10000},
```
Comment (DPG17) [inline run_addb_batch.sh:13]: Will this run just two s3 bench tests?
We may have to run multiple tests (more than these two tests)
Would developer need to modify the script (i.e add more json rows here) to add more tests?

**Review by DPG17 — APPROVED** [review]

File: run_addb_batch.sh (hunk 4 above)
Comment (dsurnin) [reply run_addb_batch.sh:13]: these tests are just an example. you should edit it write the tests you need to run. you can add any number tests

**Comment by nileshgovande** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Mini-Provisioning/job/cortx-s3server/1443/ passed

**Comment by nileshgovande** [comment]
All jobs passed - merging

**Pull request merged**